### PR TITLE
Several a11y bug fixes

### PR DESCRIPTION
**Pull request by gingi**
Fixes:

[AB#446](https://azurebatch.visualstudio.com/3426cbfe-4c9a-4da4-88df-70f025a77017/_workitems/edit/446)
[AB#451](https://azurebatch.visualstudio.com/3426cbfe-4c9a-4da4-88df-70f025a77017/_workitems/edit/451)
[AB#452](https://azurebatch.visualstudio.com/3426cbfe-4c9a-4da4-88df-70f025a77017/_workitems/edit/452)
[AB#453](https://azurebatch.visualstudio.com/3426cbfe-4c9a-4da4-88df-70f025a77017/_workitems/edit/453)
[AB#996](https://azurebatch.visualstudio.com/3426cbfe-4c9a-4da4-88df-70f025a77017/_workitems/edit/996)

### CODE REVIEW

File: src/@batch-flask/core/html-util/focus-helper.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    return false;
+}
+
+/**
+ * Determines if an HTML element is interactive.
+ *
+ * @param element An HTML element to query
+ * @returns whether the element is interactive
+ */
+export function isInteractive(element: HTMLElement): boolean {
```
Comment (dpwatrous) [inline src/@batch-flask/core/html-util/focus-helper.ts:31]: This should check if disabled="true" (disabled elements aren't focusable)

Comment (dpwatrous) [reply src/@batch-flask/core/html-util/focus-helper.ts:31]: This should probably check for `display: none` and `visibility: hidden` on the computed style.

File: src/@batch-flask/ui/callout/callout.directive.spec.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        fixture.detectChanges();
+    });
+
+    it("can be toggled by keyboard", () => {
+        const directive = testComponent.theDirective;
+        const openSpy = spyOn(directive, "open");
+        const closeSpy = spyOn(directive, "close");
+
+        input.dispatchEvent(new KeyboardEvent("keydown", { key: "Enter" }));
```
Comment (dpwatrous) [inline src/@batch-flask/ui/callout/callout.directive.spec.ts:47]: Add a test for space as well as enter?

**Review by dpwatrous — APPROVED** [review]

**Comment by cRui861** [comment]
Fixes #2634

**Pull request merged**